corehq/messaging/scheduling/forms.py

```diff
@@ -1,22 +1,169 @@
+from crispy_forms import layout as crispy
+from crispy_forms import bootstrap as twbscrispy
+from crispy_forms.helper import FormHelper
 from django import forms
+from django.forms.fields import (
+    BooleanField,
+    CharField,
+    ChoiceField,
+    MultipleChoiceField,
+)
+from django.forms.forms import Form
 from django.template import Context
 from django.template.loader import get_template
 from django.urls import reverse
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.style import crispy as hqcrispy
+from corehq.apps.translations.models import StandaloneTranslationDoc
+from corehq.apps.users.models import CommCareUser
 
 
 class MessageRecipientWidget(forms.Widget):
-    def __init__(self, domain, attrs=None, id='message-recipient'):
+    def __init__(self, domain, readonly, attrs=None, id='message-recipient'):
         super(MessageRecipientWidget, self).__init__(attrs)
         self.domain = domain
         self.id = id
         self.query_url = reverse('possible_sms_recipients', args=[self.domain])
+        self.readonly = readonly
 
     def render(self, name, value, attrs=None):
+        initial_value = []
+        if value:
+            for doc_type, doc_id in value:
+                user = CommCareUser.wrap(CommCareUser.get_db().get(doc_id))
+                initial_value.append({"id": doc_id, "name": user.raw_username})
         # TODO populate this with inital data if necessary
         return get_template('scheduling/partials/message_recipient_widget.html').render(Context({
             'id': self.id,
             'name': name,
-            'value': '',
+            'value': 'filler' if initial_value else '',
             'query_url': self.query_url,
-            'initial_data': [],
+            'initial_data': initial_value,
+            'readonly': self.readonly
         }))
+
+
+class MessageRecipientField(MultipleChoiceField):
+    def to_python(self, value):
+        if not value:
+            return []
+        return value.split(',')
+
+    def validate(self, value):
+        # TODO Will need to add more than user ids
+        # TODO batch id verification
+        for user_id in value:
+            CommCareUser.get_db().get(user_id)
```

Comment:
also should doublecheck domain here

Reply:
I needed to move this so that it had access to the domain https://github.com/dimagi/commcare-hq/pull/17722/commits/0348c1d6c426985e8f8c5bab5f973a6f30b6f29e